MoreLinq/Collections/Dictionary.cs

```diff
@@ -17,55 +17,40 @@
 
 namespace MoreLinq.Collections
 {
+    using System;
     using System.Collections.Generic;
     using System.Diagnostics.CodeAnalysis;
 
     /// <summary>
     /// A minimal <see cref="System.Collections.Generic.Dictionary{TKey,TValue}"/> wrapper that
-    /// allows null keys when <typeparamref name="TKey"/> is a
-    /// reference type.
+    /// allows a null key.
     /// </summary>
 
     // Add members if and when needed to keep coverage.
 
     sealed class Dictionary<TKey, TValue>
     {
-        readonly System.Collections.Generic.Dictionary<TKey, TValue> _dict;
-        (bool, TValue) _null;
+        readonly System.Collections.Generic.Dictionary<ValueTuple<TKey>, TValue> _dict;
 
-        public Dictionary(IEqualityComparer<TKey> comparer)
-        {
-            _dict = new System.Collections.Generic.Dictionary<TKey, TValue>(comparer);
-            _null = default;
-        }
+        public Dictionary(IEqualityComparer<TKey> comparer) =>
+            _dict = new System.Collections.Generic.Dictionary<ValueTuple<TKey>, TValue>(new ValueTupleItemComparer<TKey>(comparer));
```

Comment:
This should pass `null` if `comparer == EqualityComparer<TKey>.Default`. Both `Dictionary<,>` and `ValueTuple<>` are specifically optimized in the code and JIT for native types if default comparer is used, so adding a non-default comparer over a default comparer reduces performance.

Reply:
I wanted to come back to that, but then forgot about it completely. Thanks for pointing it out and I've addressed it with 8f19713ca487e019efeed059f0704be570949d04.